packages/ui/src/components/va-time-input/VaTimeInput.vue

```diff
@@ -189,20 +227,54 @@ export default defineComponent({
       isValid.value = true
     })
 
-    expose({
-      reset,
-      focus,
-      blur,
-    })
+    const hideDropdown = () => {
+      isOpenSync.value = false
+      focus()
+    }
+
+    const showDropdown = () => {
+      if (props.disabled || props.readonly) { return }
+      isOpenSync.value = true
+    }
+
+    const dropdownToggle = () => {
+      isOpenSync.value ? hideDropdown() : showDropdown()
+    }
+
+    const handleComponentClick = (e: any) => {
+      const id: string | undefined = e.target?.id
```

Comment:
Why do we need id? You can simply make <va-icon clear-icon @click="focus". Please, change it.

Reply:
We use the global handler to prevent the toggle dropdown on any click and execute additional logic